Sources/Amplitude/Plugins/iOS/IOSLifecycleMonitor.swift

```diff
@@ -7,131 +7,137 @@
 
 #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
 
-    import Foundation
-    import SwiftUI
-
-    class IOSLifecycleMonitor: UtilityPlugin {
-        private var application: UIApplication?
-        private var appNotifications: [NSNotification.Name] = [
-            UIApplication.didEnterBackgroundNotification,
-            UIApplication.willEnterForegroundNotification,
-            UIApplication.didFinishLaunchingNotification,
-            UIApplication.didBecomeActiveNotification,
-        ]
-        private var utils: DefaultEventUtils?
-        private var sendApplicationOpenedOnDidBecomeActive = false
-
-        override init() {
-            // TODO: Check if lifecycle plugin works for app extension
-            // App extensions can't use UIApplication.shared, so
-            // funnel it through something to check; Could be nil.
-            application = UIApplication.value(forKeyPath: "sharedApplication") as? UIApplication
-            super.init()
-            setupListeners()
+import Foundation
+import SwiftUI
+
+class IOSLifecycleMonitor: UtilityPlugin {
+    private var application: UIApplication? {
+        // TODO: Check if lifecycle plugin works for app extension
+        // App extensions can't use UIApplication.shared, so
+        // funnel it through something to check; Could be nil.
+        UIApplication.value(forKeyPath: "sharedApplication") as? UIApplication
+    }
```

Comment:
Wrapped the logic to get the `UIApplication.shared` in a computed property since `UIApplication.shared` is not available during the app startup.

Reply:
I've tried accessing the `application` instance later when the app starts and with the previous code, the instance is always `nil`.